src/langsmith/deployments-scale.mdx

```diff
@@ -0,0 +1,353 @@
+- Queue: Handles the execution of runs.
+- Redis: Handles the storage of ephemeral data about on-going runs.
+- Postgres: Handles the storage of all data, including run, thread, assistant, cron job, checkpointing and long term memory
+
+### Best practices for scaling the write path
+
+#### Enable the use of queue replicas
+
+By default, the api replica manages the queue and does not use queue replicas. You can enable the use of queue replicas by setting the `queue.enabled` configuration to `true`.
+
+```yaml
+queue:
+  enabled: true
+```
+
+This will allow the api replica to offload the queue management to the queue replicas, significantly reducing the load on the api replica and allowing it to focus on handling requests.
+
+#### Change N_JOBS_PER_WORKER based on assistant characteristics
+
+The default value of N_JOBS_PER_WORKER is 10. You can change this value to scale the maximum number of runs that can be executed at a time by a single queue replica based on the characteristics of your assistant.
+
+Some general guidelines for changing N_JOBS_PER_WORKER:
+
+- If your assistant is computationally intensive, lower N_JOBS_PER_WORKER to handle fewer concurrent runs per replica or keep it at the default.
+- If your assistant is computationally lightweight, increase N_JOBS_PER_WORKER to handle more concurrent runs per replica.
+
+There is no upper limit to N_JOBS_PER_WORKER. However, the polling algorithm queues use to pick up new runs is greedy, meaning they will try to pick up as many runs as possible and begin executing them right away. If N_JOBS_PER_WORKER is too high and traffic is burstly, you may see uneven replica utilization and potentially longer run execution times.
+
+#### Support a number of workers equal to expected throughput
+
+The more runs you execute in parallel, the more workers you will need to handle the load. There are two main parameters to scale the available workers:
+
+- number_of_queue_replicas: The number of queue replicas provisioned.
+- N_JOBS_PER_WORKER: The number of runs that can be executed at a time by a single queue replica. Defaults to 10.
+
+You can calculate the available workers with the following equation:
+```
+available_workers = number_of_queue_replicas * N_JOBS_PER_WORKER
+```
+
+Throughput is then the number of runs that can be executed per second by the available workers:
+```
+throughput_per_second = available_workers / average_run_execution_time_seconds
+```
+
+So the minimum number of workers you should provision to support your expected steady state throughput is:
+```
+number_of_queue_replicas = throughput_per_second * average_run_execution_time_seconds / N_JOBS_PER_WORKER
+```
+
+#### Configure autoscaling for bursty workloads
+
+Autoscaling is disabled by default, but should be configured for bursty workloads. Using the same calculations as above, you can determine the maximum number of workers you should allow the autoscaler to scale to based on maximum expected throughput.
```

Comment:
Commenting again. We need to distinguish which sections are applicable to SaaS, self-hosted, or both.